### PR TITLE
Algorithm challenge: Combinations

### PR DESCRIPTION
**Pull request by noelfoka**
## Algorithm challenge: Combinations

In this challenge, we wrote a function that:

- "combine" generates all possible combinations of numbers from `1` to `n` with length `k`.
- Uses a recursive helper function `generateCombinations` to generate the combinations.

### CODE REVIEW

File: algorithms/combinations.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## Context&nbsp;[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2Fnumerica-ideas%2Fcommunity%2Ftree%2Fmaster%2Falgorithms%2FCombinations&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://numericaideas.com/)
+This challenge comes from Leetcode, it's about implementation of the "combine" function, which generates all possible combinations of numbers from 1 to n with length k.
+
+**Challenge**: https://leetcode.com/problems/combinations/description/.
+
+**Solution**:
+The code provided below is a JavaScript implementation of the "combine" function, which generates all possible combinations of numbers from 1 to n with length k.
```
Comment (messaismael) [inline algorithms/combinations.md:7]: .., which return all possible combinations of `k` numbers chosen from the range `[1, n]` given two integers `n` and `k`, representing respectively the range of numbers and the length of the combinations..

File: algorithms/combinations.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  if (n < k || k < 1) return [];
+
+  var result = [];
+
+  x(result, [], 0, n, k);
+
+  return result;
+};
+
+var x = function (result, y0, x0, n, k) {
```
Comment (messaismael) [inline algorithms/combinations.md:45]: The name of this function is not meaningful enough

Comment (messaismael) [reply algorithms/combinations.md:45]: The same comment regarding its parameters:  `y0`,  `x0`

File: algorithms/combinations.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  for (var i = x0; i < n; i++) {
+    y0.push(i + 1)
+    x(result, y0, i + 1, n, k - 1);
+    y0.pop();
+  }
+};
+```
+
+By Noel Foka
```
Comment (messaismael) [inline algorithms/combinations.md:59]: The link of your GitHub profile

File: algorithms/combinations.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+## Context&nbsp;[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2Fnumerica-ideas%2Fcommunity%2Ftree%2Fmaster%2Falgorithms%2FCombinations&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://numericaideas.com/)
+This challenge comes from Leetcode, it's about implementation of the "combine" function, which generates all possible combinations of numbers from 1 to n with length k.
+
+**Challenge**: https://leetcode.com/problems/combinations/description/.
+
+**Solution**:
+The code provided below is a JavaScript implementation of the "combine" function, which generates all possible combinations of numbers from 1 to n with length k.
+
+ This is because the function takes two parameters, n and k, which represent the range of numbers and the length of the combinations, respectively. It returns an array of arrays, where each inner array represents a combination.
```
Comment (messaismael) [inline algorithms/combinations.md:9]: You can remove this sentence

**Review by messaismael — CHANGES_REQUESTED** [review]
Highlight the name of the variables, functions name, and parameters that you mention in the description.

Egg: 
The function first checks if the input is valid (`n` must be greater than or equal to `k`, and `k` must be greater than or equal to `1`). If the input is invalid, an empty array is returned.

**Review by messaismael — APPROVED** [review]

**Comment by messaismael** [comment]
@dassiorleando It's 🆗

**Review by dassiorleando — CHANGES_REQUESTED** [review]
@noelfoka could you please add the **Runtime** before your name at the end?

Example: https://github.com/numerica-ideas/community/blob/master/algorithms/maximum-subarray.md

**Comment by noelfoka** [comment]
I think it's already good

**Review by dassiorleando — APPROVED** [review]

**Comment by dassiorleando** [comment]
Great code change @noelfoka and Excellent review as usual @messaismael, I just approved it.

@messaismael do you see any possibilities of improvements for the guide that you wrote?
https://github.com/numerica-ideas/community/blob/master/algorithms/guide.md

**Pull request merged**